source/appModules/devenv.py

```diff
@@ -473,3 +486,36 @@ class IVsTextManager(comtypes.IUnknown):
     COMMETHOD([], HRESULT, 'SetTopLine',
               ( ['in'], c_int, 'iBaseLine' )),
 ]
+
+class UIATextInfoForVisualStudio( UIATextInfo):
+
+	def __init__(self,obj,position,_rangeObj=None):
+		super(UIATextInfoForVisualStudio,self).__init__(obj,position,_rangeObj)
+		self.digitsAtStartOfLineMatch = re.compile(r"^[\d]+\s")
+
+	def _getFormatFieldAtRange(self,range,formatConfig):
+		formatField = super( UIATextInfoForVisualStudio , self)._getFormatFieldAtRange(range,formatConfig)
+		currentElement = self.NVDAObjectAtStart
+		isLineNumberAtStart = self.digitsAtStartOfLineMatch.match(range.GetText(-1))

```

Comment:
My concern here is that this will match for any range, not just lines. For example, if you move through a line word by word and there's a number in the middle of the line (something as simple as "x = 10"), it will try to match a breakpoint to that number. I think we're going to need to override expand and maybe other methods so we know whether we're dealing with the start of a line.

@MichaelDCurran, any idea how we could do this nicely for UIA? _getFormatFieldAtRange could get called for multiple ranges in a single line, so I'm not even sure overriding expand is enough here. But perhaps Visual Studio only has one format range per line? I'm not sure.